compiler-plugin/src/main/kotlin/arrow/meta/plugins/proofs/phases/resolve/ProofTypeChecker.kt

```diff
@@ -41,11 +41,11 @@ private const val logTypeSize: Int = 50
 
 class ProofTypeChecker(private val compilerContext: CompilerContext) : NewKotlinTypeChecker {
 
-  override fun isSubtypeOf(p0: KotlinType, p1: KotlinType): Boolean {
-    val result = baseLineTypeChecker.isSubtypeOf(p0, p1)
-    return if (!result && !p0.isError && !p1.isError) {
-      Log.Verbose({ "ProofTypeChecker.isSubtypeOf: ${p0.toString().take(logTypeSize)} : ${p1.toString().take(logTypeSize)} -> $this" }) {
-        compilerContext.extensionProof(p0, p1) != null
+  override fun isSubtypeOf(subtype: KotlinType, supertype: KotlinType): Boolean {
```

Comment:
👍 this rename helped me to understand what's happening here. 

Maybe out of scope, but: Isn't this more like a global function we should have available in the library for all `KotlinType`s maybe as an extension? or the implementation of "subtype of" is different now depending on the type checker?

Reply:
The typechecker is instantiated by the compiler multiple times so we piggy back in every instance they create to replace it for this typechecker that gives precedence to the baseline type checker which is the original one the compiler provides in different parts.

Reply:
But the subtype relationship varies? or is this code that could be delegated to a shared ext fun?

Reply:
yes, you can delegate the body of the type checker to an ext fun but it's not necessary because that is already happening when you call `subtype.isSubtypeOf(superType)` that call that is already part of the compiler uses our injected KotlinTypeChecker so it's unnecessary wrapping IMO